### PR TITLE
Check in generated C header

### PR DESCRIPTION
**Pull request by d-e-s-o**
Please see individual commits for descriptions.

### CODE REVIEW

**Comment by d-e-s-o** [comment]
What it looks like when the C header is outdated: https://github.com/d-e-s-o/blazesym/actions/runs/4388962270/jobs/7686032079#step:5:1

File: include/blazesym.h [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+ * BlazeSymbolizer provides an interface to symbolize addresses with
+ * a list of symbol sources.
+ *
+ * Users should present BlazeSymbolizer with a list of symbol sources
+ * (`SymbolSrcCfg`); for example, an ELF file and its base address
+ * (`SymbolSrcCfg::Elf`), or a Linux kernel image and a copy of its
+ * kallsyms (`SymbolSrcCfg::Kernel`).  Additionally, BlazeSymbolizer
+ * uses information from these sources to symbolize addresses.
+ */
+typedef struct BlazeSymbolizer BlazeSymbolizer;
```
Comment (anakryiko) [inline include/blazesym.h:104]: does bindgen generate all these typedefs or it's somehow controllable? It's a bit out of style for kernel and libbpf styles, so I wonder if we can get rid of them?

**Review by anakryiko — APPROVED** [review]
It's a good change, thanks!

Let's  dedicate some time (offline) to go over all the naming choices separately and try to clean up C-side of things before it's too late to make any changes.

File: include/blazesym.h (hunk 1 above)
Comment (danielocfb) [reply include/blazesym.h:104]: I haven't found a way to customize that aspect of the generation.

**Pull request merged**